app/models/application.rb

```diff
@@ -0,0 +1,20 @@
+class Application < ApplicationRecord
+  has_many :application_pets
+  has_many :pets, through: :application_pets
+
+  validates :name, presence: true
+  validates :address, presence: true
+  validates :description, presence: true
+
+  def pet_to_adopt(pet_id) #pet_id coming through as nil, still need to figure out
+    # return if pet_id.nil?
+    wanted_pet = Pet.find_by(id: pet_id)
```

Comment:
I think this might work if you just do a find instead of a find by and put Pet.find(pet_id) - the default search value for a find is the pet id number